src/script/conversation/ConversationRepository.ts

```diff
@@ -1281,18 +1284,16 @@ export class ConversationRepository {
    * @param userId ID of service user to be removed from the conversation
    * @returns Resolves when service was removed from the conversation
    */
-  public removeService(conversationEntity: Conversation, userId: string) {
-    return this.conversation_service.deleteBots(conversationEntity.id, userId).then((response: any) => {
-      // TODO: Can this even have a response? in the API Client it look like it always returns `void`
-      const hasResponse = response?.event;
-      const currentTimestamp = this.serverTimeHandler.toServerTimestamp();
-      const event = hasResponse
-        ? response.event
-        : EventBuilder.buildMemberLeave(conversationEntity, userId, true, currentTimestamp);
-
-      this.eventRepository.injectEvent(event, EventRepository.SOURCE.BACKEND_RESPONSE);
-      return event;
-    });
+  public async removeService(conversationEntity: Conversation, userId: string): Promise<any> {
+    const response: any = await this.conversation_service.deleteBots(conversationEntity.id, userId);
+    // TODO: Can this even have a response? in the API Client it look like it always returns `void`
+    const hasResponse = response?.event;
+    const currentTimestamp = this.serverTimeHandler.toServerTimestamp();
+    const event = hasResponse
```

Comment:
I think this can be simplified 
```suggestion
    const event = response?.event || EventBuilder.buildMemberLeave(conversationEntity, userId, true, currentTimestamp);
```
and get rid of line 1290